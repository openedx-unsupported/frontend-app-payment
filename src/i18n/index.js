import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
