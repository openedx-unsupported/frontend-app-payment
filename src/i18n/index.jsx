import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
