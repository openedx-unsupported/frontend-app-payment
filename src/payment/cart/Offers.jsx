import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedNumber } from '@edx/frontend-i18n';

import LocalizedPrice from './LocalizedPrice';

function Benefit({ benefitType, benefitValue }) {
  if (benefitType === 'Percentage') {
    return <FormattedNumber value={benefitValue / 100} style="percent" />; // eslint-disable-line react/style-prop-object
  }
  return <LocalizedPrice amount={benefitValue} />;
}

Benefit.propTypes = {
  benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
  benefitValue: PropTypes.number.isRequired,
};

function Offer({
  benefitType, benefitValue, provider,
}) {
  let message = null;
  if (provider) {
    message = (<FormattedMessage
      id="payment.summary.discount.offer"
      defaultMessage="{benefit} discount provided by {provider}."
      description="A description of a discount offer applied to a basket."
      values={{
        benefit: (
          <Benefit
            benefitType={benefitType}
            benefitValue={benefitValue}
          />
        ),
        provider,
      }}
    />);
  } else {
    message = (<FormattedMessage
      id="payment.summary.discount.dynamic_offer"
      defaultMessage="{benefit} discount for your first upgrade applied."
      description="A description of a discount offer applied to a basket."
      values={{
        benefit: (
          <Benefit
            benefitType={benefitType}
            benefitValue={benefitValue}
          />
        ),
      }}
    />);
  }
  return (
    <p className="m-0 text-muted" key={`${benefitValue}-${provider}`}>
      {message}
    </p>
  );
}

Offer.propTypes = {
  benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
  benefitValue: PropTypes.number.isRequired,
  provider: PropTypes.string,
};

Offer.defaultProps = {
  provider: null,
};

export default function Offers({ offers, discounts }) {
  if ((discounts === undefined || discounts <= 0) && offers.length === 0) return null;

  return (
    <div className="summary-row">
      <p className="d-flex m-0">
        <span className="flex-grow-1">
          <FormattedMessage
            id="payment.summary.table.label.discount.total"
            defaultMessage="Discounts applied"
            description="Label for the total discount applied to an order"
          />
        </span>
        <span>
          <LocalizedPrice amount={discounts ? discounts * -1 : 0} />
        </span>
      </p>
      {offers.map(offer => (
        <Offer key={`${offer.benefitValue}-${offer.benefitType}-${offer.provider}`} {...offer} />
      ))}
    </div>
  );
}

Offers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
    benefitValue: PropTypes.number.isRequired,
    provider: PropTypes.string,
  })),
  discounts: PropTypes.number,
};

Offers.defaultProps = {
  offers: [],
  discounts: undefined,
};
