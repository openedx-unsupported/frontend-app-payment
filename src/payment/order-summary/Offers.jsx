import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedNumber } from '@edx/frontend-i18n';

import LocalizedPrice from './LocalizedPrice';

function Benefit({ benefitType, benefitValue }) {
  if (benefitType === 'Percentage') {
    return <FormattedNumber value={benefitValue / 100} style="percent" />; // eslint-disable-line react/style-prop-object
  }
  return <LocalizedPrice amount={benefitValue} />;
}

Benefit.propTypes = {
  benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
  benefitValue: PropTypes.number.isRequired,
};

function Offer({
  benefitType, benefitValue, provider,
}) {
  return (
    <p className="m-0 text-muted" key={`${benefitValue}-${provider}`}>
      <FormattedMessage
        id="payment.summary.discount.offer"
        defaultMessage="{benefit} discount provided by {provider}."
        description="A description of a discount offer applied to a basket."
        values={{
          benefit: (
            <Benefit
              benefitType={benefitType}
              benefitValue={benefitValue}
            />
          ),
          provider,
        }}
      />
    </p>
  );
}

Offer.propTypes = {
  benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
  benefitValue: PropTypes.number.isRequired,
  provider: PropTypes.string.isRequired,
};

export default function Offers({ offers, discounts }) {
  if ((discounts === undefined || discounts <= 0) && offers.length === 0) return null;

  return (
    <div className="summary-row">
      <p className="d-flex m-0">
        <span className="flex-grow-1">
          <FormattedMessage
            id="payment.summary.table.label.discount.total"
            defaultMessage="Discounts applied"
            description="Label for the total discount applied to an order"
          />
        </span>
        <span>
          <LocalizedPrice amount={discounts * -1} />
        </span>
      </p>
      {offers.map(offer => (
        <Offer key={`${offer.benefitValue}-${offer.benefitType}-${offer.provider}`} {...offer} />
      ))}
    </div>
  );
}

Offers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    benefitType: PropTypes.oneOf(['Percentage', 'Absolute']).isRequired,
    benefitValue: PropTypes.number.isRequired,
    provider: PropTypes.string.isRequired,
  })),
  discounts: PropTypes.number,
};

Offers.defaultProps = {
  offers: [],
  discounts: undefined,
};
