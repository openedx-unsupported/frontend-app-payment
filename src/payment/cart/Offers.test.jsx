import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import Offers from './Offers';

const mockStore = configureMockStore();

const baseState = { payment: { basket: {} } };

const renderWithProviders = children => renderer.create((
  <IntlProvider locale="en">
    <Provider store={mockStore(baseState)}>
      {children}
    </Provider>
  </IntlProvider>
));

describe('<Offers />', () => {
  it('renders nothing if not supplied a discount value', () => {
    const tree = renderWithProviders(<Offers />).toJSON();
    expect(tree).toBeNull();
  });

  it('renders a percentage offer', () => {
    const tree = renderWithProviders((
      <Offers
        offers={[
          { benefitValue: 50, benefitType: 'Percentage', provider: 'Pied Piper' },
        ]}
        discounts={10}
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a dynamic discount offer', () => {
    const tree = renderWithProviders((
      <Offers
        offers={[
          { benefitValue: 50, benefitType: 'Percentage', provider: null },
        ]}
        discounts={10}
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an absolute value offer', () => {
    const tree = renderWithProviders((
      <Offers
        offers={[
          { benefitValue: 10, benefitType: 'Absolute', provider: 'Pied Piper' },
        ]}
        discounts={10}
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
