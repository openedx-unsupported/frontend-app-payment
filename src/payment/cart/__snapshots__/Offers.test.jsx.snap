// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offers /> renders a dynamic discount offer 1`] = `
<div
  className="summary-row"
>
  <p
    className="d-flex m-0"
  >
    <span
      className="flex-grow-1"
    >
      Discounts applied
    </span>
    <span>
      -$10.00
    </span>
  </p>
  <p
    className="m-0 text-muted"
  >
    50%
     discount for your first upgrade applied.
  </p>
</div>
`;

exports[`<Offers /> renders a percentage offer 1`] = `
<div
  className="summary-row"
>
  <p
    className="d-flex m-0"
  >
    <span
      className="flex-grow-1"
    >
      Discounts applied
    </span>
    <span>
      -$10.00
    </span>
  </p>
  <p
    className="m-0 text-muted"
  >
    50%
     discount provided by Pied Piper.
  </p>
</div>
`;

exports[`<Offers /> renders an absolute value offer 1`] = `
<div
  className="summary-row"
>
  <p
    className="d-flex m-0"
  >
    <span
      className="flex-grow-1"
    >
      Discounts applied
    </span>
    <span>
      -$10.00
    </span>
  </p>
  <p
    className="m-0 text-muted"
  >
    $10.00
     discount provided by Pied Piper.
  </p>
</div>
`;
